src/catalog.h

```diff
@@ -873,10 +875,17 @@ typedef struct FormData_continuous_agg
 	NameData partial_view_schema;
 	NameData partial_view_name;
 	/*
-	 * Don't access bucket_width directly to determine the width of the bucket.
-	 * Use corresponding procedures instead:
+	 * bucket_width is BUCKET_WIDTH_VARIABLE (see continuous_agg.h) for buckets
+	 * with variable size - monthly buckets, buckets with timezone, etc. For such
+	 * buckets the information about the bucketing function is stored in
+	 * _timescaledb_catalog.continuous_aggs_bucket_function.
+	 *
+	 * When possible, don't access bucket_width directly. Use corresponding
```

Comment:
I'm wondering if we can just replace all of this with some sort of function call, sort of like we do with partitioning time function. So that we aren't calculating any of this ourselves and instead just calling a function that is a function of a certain type?